### DISCUSSION TITLE
Custom database solution

### DISCUSSION THREAD

**Discussion by whalemare** [appwrite/appwrite#1486] [Q&A] [answered]
Can I use appwrite with my own database implementation service? 

I have a lot of code that already works with my existed database (searches, filters and etc.). I want to use appwrite, but don't want code again the same logic from my databases queries to appwrite collections schema. 

How can it be done?

**Comment by maeddin** [comment, marked as answer]
I would suggest having a separate database server application with its own API that uses JWT tokens from Appwrite for authentication. Depending on what language you use, you can also use an official server SDK from Appwrite for this.
For example, I have already written a project with Java and MongoDB and added Appwrite for authentication and storage afterwards.

**Reply by whalemare** [reply]
Yes, after some digging in this direction I came to the same conclusion. Thanks
